processing/src/main/java/org/apache/druid/segment/column/ComplexColumn.java

```diff
@@ -19,45 +19,29 @@
 
 package org.apache.druid.segment.column;
 
+import org.apache.druid.guice.annotations.ExtensionPoint;
 import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
 import org.apache.druid.segment.ColumnValueSelector;
 import org.apache.druid.segment.ObjectColumnSelector;
-import org.apache.druid.segment.data.GenericIndexed;
 import org.apache.druid.segment.data.ReadableOffset;
 
 import javax.annotation.Nullable;
 
 /**
-*/
-public class ComplexColumn implements BaseColumn
+ */
+@ExtensionPoint
+public interface ComplexColumn extends BaseColumn
```

Comment:
Would you please add some javadoc?

Reply:
added

Reply:
Oh sorry, I should mention this in the first place, but would you please add simple descriptions for methods as well? Since this is an extension point now, it would make it clear what their implementation requires.

Reply:
added more to the best of my knowledge :)